### PR TITLE
Fix metadata table conflict inserting GVR config

### PR DESCRIPTION
**Pull request by jframe**
The genesis validators root is insert on first received request. If multiple requests are received at the same time on an empty database some of those requests could fail due to a constraint ensuring that only one GVR should be in the metadata table.

This change will retry the check and insert logic for the GVR if we get an insert error as it is quite likely that just retrying will fix the error.

### CODE REVIEW

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/validator/GenesisValidatorRootValidator.java [hunk 1]
```diff
@@ -15,33 +15,70 @@
...
+            metadataDao.insertGenesisValidatorsRoot(handle, genesisValidatorsRoot);
+          }
+          return isValidGvr;
+        });
+  }
+
+  private <V> V retry(final Callable<V> callable) {
+    try {
+      return callable.call();
+    } catch (Exception e) {
```
Comment (usmansaleem) [inline slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/validator/GenesisValidatorRootValidator.java:65]: nit: final

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbTransactionRetryer.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      }
+    }
+    throw new IllegalStateException(
+        "Transaction max retries " + maxRetries + " reached. Not retrying transaction");
+  }
+
+  private void retrySleep() {
+    try {
+      final int jitter = random.nextInt(maxJitterMs);
+      Thread.sleep(retryMs + jitter);
```
Comment (usmansaleem) [inline slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbTransactionRetryer.java:62]: ouch ...

Comment (usmansaleem) [reply slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbTransactionRetryer.java:62]: may I suggest something like this instead? https://github.com/jhalterman/failsafe

**Review by usmansaleem — APPROVED** [review]
LGTM - suggestion around using Failsafe API which provides a better DSL to deal with retry and/or circuitbreaker pattern.

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbTransactionRetryer.java (hunk 2 above)
Comment (jframe) [reply slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbTransactionRetryer.java]: nice suggestion. done

**Pull request merged**